core/src/main/java/org/elasticsearch/index/mapper/core/KeywordFieldMapper.java

```diff
@@ -128,6 +120,8 @@ public KeywordFieldMapper build(BuilderContext context) {
                 } else if (propName.equals("ignore_above")) {
                     builder.ignoreAbove(XContentMapValues.nodeIntegerValue(propNode, -1));
                     iterator.remove();
+                } else if (TypeParsers.parseNorms(builder, propName, propNode, parserContext)) {
```

Comment:
If you like it better, I can go the other route and upgrade the norms settings in StringFieldMapper.Parser so that neither text nor keyword have to know about the old norms settings.